test/controlplane/ciliumnetworkpolicies/status_updates_gc.go

```diff
@@ -243,28 +243,7 @@ func init() {
 		// We only need to test the last k8s version
 		test := suite.NewControlPlaneTest(t, "cnp-status-update-control-plane", k8sVersions[len(k8sVersions)-1])
 
-		// When running with GC disabled, the Nodes Status updates should not be deleted.
-		test.
-			UpdateObjects(initialObjects...).
-			SetupEnvironment().
-			// check that CNPs contain status updates info before starting agent and operator
-			Eventually(func() error { return validateCNPs(test) }).
-			StartAgent(func(_ *option.DaemonConfig) {}).
-			StartOperator(
-				func(operatorCfg *operatorOption.OperatorConfig) {
-					operatorCfg.SkipCNPStatusStartupClean = true
-				},
-				func(vp *viper.Viper) {
-					vp.Set(operatorApi.OperatorAPIServeAddr, "localhost:0")
-				},
-			).
-			Eventually(func() error { return validateCNPs(test) }).
-			StopAgent().
-			StopOperator().
-			DeleteObjects(initialObjects...).
-			ClearEnvironment()
-
-		// When running with GC enabled, the Nodes Status updates should eventually be deleted.
+		// CNP nodes status updates are deprecated and should be deleted.
```

Comment:
This patch (`controlplane: remove obsolete CNPNodeStatusGC test`) applies as is only to `main` - let's drop it from v1.15 as here there is actually still a difference between the two test runs. Unchanged, these are still racy, however. @gandro If you're up for it, you could change the test in `controlplane: wait for watcher establishment` similar to the other changes; that is, establish and wait for the relevant watcher. Otherwise just drop this commit and I can prepare the change.

Reply:
Makes sense! I've dropped `controlplane: remove obsolete CNPNodeStatusGC test` and changed the CNP status GC test in `controlplane: wait for watcher establishment` - but don't feel very confident since it look sufficiently different from the other changes. I'd appreciate a review

Reply:
Yep, perfect.